cylc/flow/task_events_mgr.py

```diff
@@ -1293,3 +1317,12 @@ def unset_waiting_event_timer(self, id_key):
         """
         self._event_timers[id_key].unset_waiting()
         self.event_timers_updated = True
+
+    def reset_bad_hosts(self):
+        """Periodically clear bad_hosts list.
+        """
+        LOG.info(
+            'Periodic clearing of bad hosts which were: '
+            f'{self.bad_hosts}'
+        )
+        self.bad_hosts.clear()
```

Comment:
We should avoid logging this if the bad hosts is empty, I think this'll do it:

```suggestion
        if self.bad_hosts:
            LOG.info(
                'Periodic clearing of bad hosts which were: '
                f'{self.bad_hosts}'
            )
            self.bad_hosts.clear()
```